src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "panda_torque_mpc/obstacle_params_parser.h"
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+
+#include <ros/ros.h>
+#include <yaml-cpp/yaml.h>
+
+namespace panda_torque_mpc {
+
+  ObstacleParamsParser::ObstacleParamsParser(
+      const std::shared_ptr<ros::NodeHandle> &nh,
+      const boost::shared_ptr<pinocchio::Model> &pin_model,
+      const boost::shared_ptr<pinocchio::GeometryModel> &collision_model)
+      : nh_(nh), pin_model_(pin_model), collision_model_(collision_model){};
+
+  ObstacleParamsParser::~ObstacleParamsParser() {};
+
+void ObstacleParamsParser::addCollisions()
+{
+  // First adding the obstacles to the geometry model
+ int obs_idx = 0;
+ while (nh_->hasParam("obstacle_" + std::to_string(obs_idx))) {
+
+  std::string obstacle_name;
+
+  obstacle_name = "obstacle_" + std::to_string(obs_idx);
+  std::string type;
+
+  nh_->getParam(obstacle_name + "/type", type);
+
+  Eigen::VectorXd pose;
+  nh_->getParam(obstacle_name + "/pose", pose);
+
+  switch (type) {
+    case "sphere":
+      double radius;
+      nh_->getParam(obstacle_name + "/radius", radius);
+      addSphere(obstacle_name, radius, pose);
+      break;
+
+    case "box":
+      double x; double y; double z;
+      nh_->getParam(obstacle_name + "/x", x);
+      nh_->getParam(obstacle_name + "/y", y);
+      nh_->getParam(obstacle_name + "/z", z);
+      addBox(obstacle_name, x, y, z, pose);
+      break;
+
+    case "capsule":
+      double radius; 
+      double halfLength;
+      nh_->getParam(obstacle_name + "/radius", radius);
+      nh_->getParam(obstacle_name + "/halfLength", halfLength);
+      addCapsule(obstacle_name, radius, halfLength, pose);
+      break;
+    }
+  obs_idx += 1;
+ }
```

Comment:
```suggestion
  nh_->getParam(obstacle_name + "/rotation", rotation_vect);
  // convert rotation to eigen
  
  pinocchio::GeometryObject::CollisionGeometryPtr geometry;
  switch (type) {
    case "sphere":
      double radius;
      nh_->getParam(obstacle_name + "/radius", radius);
      geometry = pinocchio::GeometryObject::CollisionGeometryPtr(new hpp::fcl::Sphere(radius));
      break;

    case "box":
      double x; double y; double z;
      nh_->getParam(obstacle_name + "/x", x);
      nh_->getParam(obstacle_name + "/y", y);
      nh_->getParam(obstacle_name + "/z", z);
      // geometry = ...
      break;

    case "capsule":
      double radius; 
      double halfLength;
      nh_->getParam(obstacle_name + "/radius", radius);
      nh_->getParam(obstacle_name + "/halfLength", halfLength);
      // geometry = ...
      break;
    }
  pinocchio::SE3 obstacle_pose(Eigen::Quaterniond(1., 0., 0., 0.), translation);
  pinocchio::GeometryObject obstacle(name, 0, 0, geometry, obstacle_pose);
  collision_model_->addGeometryObject(obstacle);
  obs_idx += 1;
 }
```

This way, you can remove code repeatability from the helper functions. Helper functions are a good idea in general, but in this case they introduced lots of repeated lines of code.